### PR TITLE
Run workflows on pull request

### PR DESCRIPTION
**Pull request by efirs**

### CODE REVIEW

File: .github/workflows/go-test.yaml [hunk 1]
```diff
@@ -1,7 +1,10 @@
 name: go-test
 on:
+  pull_request:
   push:
-  workflow_call:
+    branches:
```
Comment (adilansari) [inline .github/workflows/go-test.yaml:5]: Instead you can make your release wflow to require test and lint workflows

**Review by adilansari — DISMISSED** [review]

File: .github/workflows/go-test.yaml (hunk 1 above)
Comment (adilansari) [reply .github/workflows/go-test.yaml:5]: https://github.com/tigrisdata/tigris-client-python/actions/runs/5063675766

Comment (efirs) [reply .github/workflows/go-test.yaml:5]: Updated, check it out.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -5,7 +5,13 @@ on:
       - "release"
       - "next"
 jobs:
+  test:
+    uses: ./.github/workflows/go-test.yaml
```
Comment (adilansari) [inline .github/workflows/release.yaml:9]: You'd also need a 'workflow_call:' in go-test.yml. Check python ci.yml for reference.

Comment (efirs) [reply .github/workflows/release.yaml:9]: ah ,right.

Comment (efirs) [reply .github/workflows/release.yaml:9]: updated.

**Review by adilansari — APPROVED** [review]

**Pull request merged**